### ISSUE TITLE
Download of episodes (files+images) fails for specific podcast

### ISSUE THREAD

**Issue by Torben2000** [SamTV12345/PodFetch#492]
### Describe the bug

For [this podcast](https://podcast.hr.de/hr2_maerchen_verbrechen/podcast.xml) the download of all episode images fails. This also seems to cause the download of the files itself to fail. 
Example log entry:

```
2023-12-23T20:00:00 ℹ️ - Downloading podcast episode: Grimms Märchen & Verbrechen, Staffel 3  - Der 12. Fall: "Die drei Feldscherer" (1/3)
thread '<unnamed>' panicked at src/service/download_service.rs:63:14:
called `Result::unwrap()` on an `Err` value: reqwest::Error { kind: Builder, source: RelativeUrlWithoutBase }```
```

Additional info: In the database some fields of the eipsodes contain unusual content:
image_url: `ui/default.jpg` (although correctly filled in podcast xml)
duration: `0` (although filled in podcast xml)
guid: URL of the file, e.g. `https://mp3podcasthr-a.akamaihd.net/mp3/podcast/hr2_maerchen_verbrechen/hr2_maerchen_verbrechen_20231222_99775242.mp3` (this is really the value in the podcast xml, but still strange)

Playing the episodes directly via UI works. Also my other podcasts work.

### Reproduction

Add specific podcast and try to download any episode

### System Info

```Shell
Running via Docker with Postgres and default config (just changed the folder on the host)
```

### Used Package Manager

n/a

### Validations

- [X] Follow our [Code of Conduct](https://github.com/SamTV12345/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/SamTV12345/.github/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by allthesebugsv2** [comment]
Have the same problem; most podcasts work fine but quite a few show the issue described here.

For example:

- https://podcast.hr.de/derTag/podcast.xml
- https://podcast.hr.de/hr-info_gespraech/podcast.xml
- https://podcast.hr.de/freiheit-deluxe/podcast.xml
- https://podcast.hr.de/hr-info_reportage/podcast.xml

Common denominator is quite clearly good old HR.

**Comment by SamTV12345** [comment]
I'll have a look at it tomorrow.

**Comment by SamTV12345** [comment]
Sorry I didn't have time lately. This is fortunately fixed.

**Comment by Torben2000** [comment]
Thanks a lot for the fix! It really improved the situation and mp3 files are downloaded. :-)

But I am really confused because the fix description suggests that the episodes do not contain images. That is not true for the podcast I tried out. All episodes have a correctly filled `itunes:image` tag (at least as far as I can see). So probably something else is still broken...

**Comment by SamTV12345** [comment]
Oh sorry. I should have checked out your podcast. The other ones didn't have an image so I assumed it was fixed.

**Comment by jpeeler** [comment]
I don't want to add noise to this issue... but I'm getting the same error on all podcasts. I'm wondering if it's a network misconfiguration. Both the album art for each episode and the data (download to server) doesn't work.

I'm running in podman with this line:
```
podman run --rm -v /home/jpeeler/container-data/podfetch/podcasts:/app/podcasts:z -v /hom
e/jpeeler/container-data/podfetch/db:/app/db:z -it -p 8000:8000 samuel19982/podfetch:latest
```

**Comment by SamTV12345** [comment]
Thanks for raising noise to this. This does seem like a problem with the library I am using.

**Comment by SamTV12345** [comment]
I'll create a workaround tomorrow so you can continue using PodFetch and still have the original images locally.

**Comment by jpeeler** [comment]
I added one of the podcasts above and it did seem to work. (Unrelated, perhaps the cloud download icon should be immediately updated upon automatic download. I clicked one while it was actually already downloading in the background due to the initial episodes being downloaded on a new feed.)

However, the feeds that I'm using do not work:
https://feed.podbean.com/thebiblerecap/feed.xml
https://feeds.simplecast.com/96_xs9dX

If it wasn't clear from my aside comment, these were both attempted on a fresh db.
